micro_sam/training/training.py

```diff
@@ -71,38 +73,44 @@ def check_instance_channel(instance_channel):
                 "All values in the target channel with the instance segmentation must be integer."
             )
 
-    n_channels_y = y.shape[1]
-    if with_segmentation_decoder:
-        if n_channels_y != 4:
-            raise ValueError(
-                "Invalid number of channels in the target data from the data loader. "
-                "Expect 4 channel for training with an instance segmentation decoder, "
-                f"but got {n_channels_y} channels."
-            )
-        check_instance_channel(y[:, 0])
-
-        targets_min, targets_max = y[:, 1:].min(), y[:, 1:].max()
-        if targets_min < 0 or targets_min > 1:
-            raise ValueError(
-                "Invalid value range in the target data from the value loader. "
-                "Expect the 3 last target channels (for normalized distances and foreground probabilities)"
-                f"to be in range [0.0, 1.0], but got min {targets_min}"
-            )
-        if targets_max < 0 or targets_max > 1:
-            raise ValueError(
-                "Invalid value range in the target data from the value loader. "
-                "Expect the 3 last target channels (for normalized distances and foreground probabilities)"
-                f"to be in range [0.0, 1.0], but got max {targets_max}"
-            )
-
-    else:
-        if n_channels_y != 1:
-            raise ValueError(
-                "Invalid number of channels in the target data from the data loader. "
-                "Expect 1 channel for training without an instance segmentation decoder,"
-                f"but got {n_channels_y} channels."
-            )
-        check_instance_channel(y)
+    name = "" if name is None else f"'{name}'"
+    for x, y in tqdm(loader, desc=f"Verifying labels in {name} dataloader"):
```

Comment:
I am a bit hesitant to do this by default. If we have a large data loader it may delay the start of training quite a bit. Let's discuss on Monday. I think the best solution would be to define a number of samples that will be checked to make this flexible.

Reply:
I exposed an additional arguments which controls whether we check labels for the entire dataloader or for the first n samples. I have left the current defaults for it to `None`, that corresponds to checking the entire dataloader.